docs/aws/audit/dynamodbmonitoring/rules/ddb_table_autoscaling_enabled.mdx

```diff
@@ -24,6 +24,285 @@ HIPAA, NIST, AWSWAF
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to list all the DynamoDB tables in your AWS account. You can do this by using the AWS CLI command `aws dynamodb list-tables`. This command will return a list of all the DynamoDB tables in your account.
+
+2. Once you have the list of tables, you can check the autoscaling settings for each table. This can be done using the AWS CLI command `aws application-autoscaling describe-scalable-targets --service-namespace dynamodb`. This command will return a list of all the scalable targets for DynamoDB in your account.
+
+3. From the output of the above command, you need to check the `ScalableDimension` field. If the value of this field is `dynamodb:table:ReadCapacityUnits` or `dynamodb:table:WriteCapacityUnits`, it means that autoscaling is enabled for read capacity or write capacity respectively.
+
+4. Finally, you need to check the `MinCapacity` and `MaxCapacity` fields. If the `MinCapacity` is not 0 or the `MaxCapacity` is not a high enough value, it means that autoscaling may not be properly configured.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. This will allow you to interact with AWS services including DynamoDB. You can install it using pip:
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and establish a session with AWS using your access keys:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2' # specify the region you are interested in
+)
+```
+
+3. Create a DynamoDB resource object using the session object. Then, iterate over all the tables in your DynamoDB and check if autoscaling is enabled:
+
+```python
+dynamodb = session.resource('dynamodb')
+
+for table in dynamodb.tables.all():
+    try:
+        response = client.describe_table(
+            TableName=table.name
+        )
+        if 'ProvisionedThroughput' in response['Table']:
+            if response['Table']['ProvisionedThroughput']['ReadCapacityUnits'] != 'auto' or response['Table']['ProvisionedThroughput']['WriteCapacityUnits'] != 'auto':
+                print(f"Autoscaling is not enabled for table {table.name}")
+    except Exception as e:
+        print(f"Error checking table {table.name}: {e}")
+```
+
+4. The above script will print out the names of all DynamoDB tables that do not have autoscaling enabled. If no output is produced, that means all tables have autoscaling enabled. If there is an error while checking a table, the error message will be printed.
+</Accordion>
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>